checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,314 @@
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import alluxio.PropertyKey;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.SparkConf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import scala.Serializable;
+import scala.Tuple2;
+
+import java.io.File;
+import java.io.PrintWriter;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import java.text.SimpleDateFormat;
+
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * The Spark integration checker includes a spark job to test the integration of Spark with Alluxio.
+ * This class will be triggered in client and cluster mode through spark-submit.
+ *
+ * This checker requires a running Spark cluster, but does not require a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized in Spark
+ * driver and executors.
+ */
+public class SparkIntegrationChecker implements Serializable{
+  private static final long serialVersionUID = 1106074873546987859L;
+
+  @Parameter(names = {"--partitions"}, description = "The user defined partition number, "
+      + "by default the value is 10")
+  private int mPartitions;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkIntegrationChecker.class);
+  private List<Tuple2<Status, String>> mSparkJobResult = null;
+  private boolean mAlluxioHAMode = false;
+  private String mZookeeperAddress = "";
+
+  /** The Spark driver and executors performIntegrationChecks results. */
+  private enum Status {
+    FAIL_TO_FIND_CLASS, // Spark driver or executors cannot recognize Alluxio classes
+    FAIL_TO_FIND_FS,    // Spark driver or executors cannot recognize Alluxio filesystem
+    FAIL_TO_SUPPORT_HA,    // Spark driver cannot support Alluxio-HA mode
+    SUCCESS;
+  }
+
+  /**
+   * Implements Spark with Alluxio integration checker.
+   *
+   * @param sc current JavaSparkContext
+   * @param printWriter save user-facing messages to a generated file
+   * @return performIntegrationChecks results
+   */
+  private Status run(JavaSparkContext sc, PrintWriter printWriter) {
+    // Checks whether Spark driver can recognize Alluxio classes and filesystem
+    Status driverStatus = performIntegrationChecks();
+    String driverAddress = sc.getConf().get("spark.driver.host");
```

Comment:
@aaudiber I am able to get the Spark driver host and port from Spark conf, but I cannot get the spark executors hosts from Spark conf. As a result, for Spark executors, I still use the old way(InetAddress.getLocalHost().getHostAddress()) to get address